### PR TITLE
Eswords autoblock unarmed attacks

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
You can't grab a lightsaber with your bare hands

EDIT: I could compromise and just give it a bonus to blocking unarmed if that'd make everyone happier

### CODE REVIEW

File: code/game/objects/items/weapons/melee/energy.dm [hunk 1]
```diff
@@ -62,8 +62,11 @@
 	if(item_color == null)
 		item_color = pick("red", "blue", "green", "purple")
 
-/obj/item/weapon/melee/energy/sword/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance)
+/obj/item/weapon/melee/energy/sword/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance, damage, attack_type)
 	if(active)
+		if(attack_type == UNARMED_ATTACK)
+			owner.visible_message("[attack_text] thinks better of trying to grab [src] with their bare hands!")
```
Comment (Anonus) [inline code/game/objects/items/weapons/melee/energy.dm]: touch not grab
this triggers on more than just disarm, yes?

File: code/game/objects/items/weapons/twohanded.dm [hunk 2]
```diff
@@ -246,8 +246,11 @@
 	else
 		user.adjustStaminaLoss(25)
 
-/obj/item/weapon/twohanded/dualsaber/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance)
+/obj/item/weapon/twohanded/dualsaber/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance, damage, attack_type)
 	if(wielded)
+		if(attack_type == UNARMED_ATTACK)
+			owner.visible_message("[attack_text] thinks better of trying to grab [src] with their bare hands!")
```
Comment (Anonus) [inline code/game/objects/items/weapons/twohanded.dm]: see above

File: code/game/objects/items/weapons/melee/energy.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/game/objects/items/weapons/melee/energy.dm]: Disarm/grab/punch

Hug bypasses it

Comment (Anonus) [inline code/game/objects/items/weapons/melee/energy.dm]: well you're not trying to grab it if you're punching, are you?

**Comment by WJohn** [comment]
Yay now you can't bullshit RNG disarm nuke ops!

**Comment by Shapsy** [comment]
>You can't grab a lightsaber with your bare hands

Not with that attitude

**Comment by Bawhoppen** [comment]
Make it so it does as much damage as getting hit but you can still disarm it potentially?

**Comment by GunHog** [comment]
Good idea, "meh" execution. Speesemen are stupid, clicking an active light bulb without gloves will cause the character to burn his hand on the light. As such, *attempting to attack/disarm someone with an esword should cause him to burn his hands on it!*

**Comment by Bawhoppen** [comment]
Well it's a lightsaber that doesn't actually do much damage so why would it do any more to your hand from touching it then getting stabbed in the chest with it

**Comment by Durkel** [comment]
Traitor: It's over assistant! I have the esword!
Assistant: You underestimate my rng!
Assistant: AAHHHHH!!
(the assistant gets cut to pieces)
Assistant: hnnng fucking kor.

**Comment by YotaXP** [comment]
Grabbing the sword?  I thought disarming was trying to shove someone down.

**Comment by KorPhaeron** [comment]
I'm not making it hit people because that'd require more arguements in both check_shields and hit_reaction

**Comment by Jazaen** [comment]
>Shove a guy that holds what basically is a blade of NOPE in front of him
This is a pretty nice buff to the esword, maybe bringing it up a little closed to the current favourite of Holoparasite. At least now you won't be beaten by single assistant and RNG!

**Comment by Zilenan91** [comment]
THANK YOU. Eswords are no longer a joke because of bullshit RnG.

**Comment by KorPhaeron** [comment]
> maybe bringing it up a little closed to the current favourite of Holoparasite. 

http://www.ss13.eu/tgdb/tg/stats_2015-10-29_to_2015-11-28.html#uplinkitems

Swords are bought more than parasites.

**Comment by Bawhoppen** [comment]
Not an excuse

**Comment by Zilenan91** [comment]
They did get a 50% price reduction recently so they're now extremely cost effective, so that could explain it.

**Comment by KorPhaeron** [comment]
>They did get a 50% price reduction recently so they're now extremely cost effective, so that could explain it.

That wasn't merged and this is data from last month

**Comment by Zilenan91** [comment]
I'm not sure if that source is entirely correct.... http://i.imgur.com/aPtIDEh.png?1

**Comment by Jazaen** [comment]
Just make edaggers disarmable. Blade or double blade is one thing, but a dagger is somehting else.

**Comment by Zilenan91** [comment]
Ehh, I wouldn't mind seeing edaggers be disarmable as they are right now. They'd be way way too cost effective if they couldn't be memed on by disarm intent. Regular eswords need it bad though.

**Comment by KorPhaeron** [comment]
@Zilenan91 

It was bought X number of times in Y number of rounds. There is more than one traitor per round, so it was on average bought more than once per round to reach that number

**Comment by KorPhaeron** [comment]
>Just make edaggers disarmable. Blade or double blade is one thing, but a dagger is somehting else.

They are not touched in this PR

**Comment by Lumipharon** [comment]
isn't this effectively a ggnore to anyone without a gun or stun baton etc?

I'm all for making it harder to rng disarm an e-sworder, but this is pretty eh.

**Comment by PKPenguin321** [comment]
:-1:

**Comment by CPTANT** [comment]
:+1:

**Comment by PKPenguin321** [comment]
:-1:

**Comment by Zilenan91** [comment]
:+1:

**Comment by PKPenguin321** [comment]
:-1:

**Comment by ike709** [comment]
:+1: merry christmas to you too kor

**Comment by PKPenguin321** [comment]
:-1: normal winter to you kor

**Comment by tkdrg** [comment]
stop spamming

**Comment by KazeEspada** [comment]
:+1:

**Comment by PKPenguin321** [comment]
no but seriously :-1: 
this will make it a little too good against general crew
the only ways you could beat this are if you have something to slip people with (pretty much only the clown/validhunters have this), a stunprod for validhunting (which is shitty), or if you're security (which makes up a minority of the crew)
i dunno

**Comment by Zilenan91** [comment]
Just throw stuff or punch them, this only counts disarms. You can still absolutely meme on them if you get a weaken.

**Comment by Lumipharon** [comment]
@Zilenan91 

>Disarm/grab/punch

Straight from kor

**Comment by Niknakflak** [comment]
I don't understand.  You can't punch someone with an E-sword?  It's like your punching through the E-sword, you are punching the guy as he swings (clicks) the E-sword at you and you manage to avoid it while punching him, not his sword.  Same goes for disarm. You can disarm someone holding a knife by hitting their arms and even then disarm is also pushing sometimes, which also doesn't have anything to do with "touching" the E-sword.  Grabs are also under the same boat.  You are grabbing the guy, not the sword.  Why does the sword even fucking matter?  Anyone who grabs a guy with an E-sword is going to get chopped to pieces but that doesn't mean you can't grab a guy with a sword.

:-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1:

**Comment by Zilenan91** [comment]
This is purely to stop people who spent nearly half their TCs on a sword from getting memed on by RnG disarm pushes, most likely.

But yeah punches should be able to go through this, I don't know why they don't.

**Comment by KorPhaeron** [comment]
I'll change the message to something else if it bothers you, but this is about making the esword stronger more than any DEEP LORE

**Comment by Niknakflak** [comment]
It's not even fucking lore, it's just logic.  Because they have a sword out means they are fucking invincible to being fucking touched? How does that make sense?

**Comment by KorPhaeron** [comment]
Okay imagine if someone was holding a sword in front of them ~3 feet long (plus the distance given by holding the arms out from their body)

Now imagine trying to touch them with your bare hands past their sword without getting cut.

Now imagine the sword is weightless and they have perfect control over it because its a magic beam of energy that fucks you up they don't even have to swing it hard.

**Comment by Niknakflak** [comment]
You can grab their hands and prevent them from moving the sword, kind of how you would grab someone's hands if they had a knife.

You could grab them if their back was turned.
You could punch them if their back was turned.
You could punch them while they swing (click at you) their sword

Direction matters, it's not as clear cut and dry.  Code limitations and blah blah aside, it's dumb for punching and grabbing at least.

**Comment by Zilenan91** [comment]
For real though punches should be able to go through these. I'm all for disarms not going through, but punches no.

**Comment by KorPhaeron** [comment]
>kind of how you would grab someone's hands if they had a knife.

Except a knife isn't 3 feet long. I promise you that trying to punch a guy with a sword is not realisitc.

>directional combat

We don't have that and I'm not adding it either.

>realism arguments in general

You can already deflect bullets with the sword

**Comment by Niknakflak** [comment]
You don't have to add it just don't cut out a situation simply because it doesn't exist.  Punching should go through. You are implying that every situation should take on the default situation of being face to face.  It's going to be a pain in the dick dealing with people with E-swords now.  They are now cheap and stupid due to this.  Stop giving in to the dumb star wars crave.  E-swords were fine as is.

**Comment by KorPhaeron** [comment]
> Stop giving in to the dumb star wars crave

This actually has nothing to do with star wars and everything to do with people have been complaining for literally years that eswords just get you disarmed.

>E-swords were fine as is.

Spending half your points on a melee only weapon that is intensely vulnerable in melee being fine is debatable.

**Comment by Niknakflak** [comment]
I've just seen so much murderbone with them.  It is true they always end in disarms but at least let punches through.   Coudn't hurt to see how this changes things anyway.  If it makes everything awful, revert is always (lmao) an option

**Comment by Zilenan91** [comment]
But yeah seriously I can't enunciate this enough punches need to go through.

**Comment by KazeEspada** [comment]
Floor tiles go through and are easy to get.
Punches are just another excuse for RNG stun. So punches are bad. :+1:

**Comment by Lumipharon** [comment]
glad everyone is a powergamer who runs around with stacks of floortiles in case someone pulls a sword on them right?

**Comment by Zilenan91** [comment]
You could always make a spear and throw that I guess. Irregardless, punches need to go through this.

**Comment by KazeEspada** [comment]
Just played a game where I was disarmed three times before finally getting my weapon jacked and then of course killed. A bunch of assholes spamming disarm is a good fighting system right?

**Comment by Bawhoppen** [comment]
until we remove stun combat it is the best solution

**Comment by PKPenguin321** [comment]
basically the reason i don't like this is because unless you're carrying some kind of powergamey gear (stunprod, soap, 60 floortiles) or you're security (which is a minority group on the station), there's basically no way of fighting these short of grabbing a fire extinguisher and relying on RNG to stun them before they inevitably stun you with their 40% knockdown chance. that basically means that a dude with a sword can waltz into medbay and cut up every guy there and they can hardly do shit about it unless they're deliberately carrying around the counters i mentioned earlier (which is shitty and shouldn't be encouraged).

**Comment by Lumipharon** [comment]
a guy running up to you with a e-sword and you can literally do nothing to stop him what so ever because your hands just bounce off his face is a good fighting system right?

**Comment by Zilenan91** [comment]
Allow punches, don't allow disarms.

**Comment by PKPenguin321** [comment]
punches have a 1 in 10 chance to weaken
extinguishers have a 1 in 5 chance to knockdown (if i remember the knockdown formula right), with 1 in 10 to KO

eswords have a 2 in 5 chance to knockdown and 3 in 10 to KO

basically if you allow punches, you're still playing the RNG game, it's just massively swayed in the swordguy's favor so that he wins basically every time if he gets the first hit or two

i say just close the PR entirely. disarms have like a 66% chance to make the swordguy drop his weapon, but the sword has like a 40% chance to just knock the dude down anyways. even if you get disarmed, you can pick your sword back up or disarm the guy back. it works fine as-is

**Comment by Lumipharon** [comment]
Nothing wrong with reducing the ease with which unarmed people can disarm e-sworders.
We just don't want to make them literally helpless unless they're packing powergamer shit.

**Comment by Zilenan91** [comment]
Or the disarm guy shits on the sword guy with a push and kills him instantly with no effort required

**Comment by Bawhoppen** [comment]
>ignoring stunprod bullshit but thinks disarm is the issue

**Comment by PKPenguin321** [comment]
>Nothing wrong with reducing the ease with which unarmed people can disarm e-sworders.
We just don't want to make them literally helpless unless they're packing powergamer shit.

100% agree

>Or the disarm guy shits on the sword guy with a push and kills him instantly with no effort required

If you push this PR, it'll be "The esword guy shits on the crewman and kills him instantly with no effort required" because there will be no counter. If you allow punching, it will be "Or the punching guy weakens the sword guy and kills him instantly with no effort required."
Besides, assistants killing nuke ops with their own weapons has been a part of the game for ages, and it's worked fine so far. If you're that salty about getting disarmed, carry a taser or something next time.

**Comment by KorPhaeron** [comment]
PK has convinced me

**Comment by PKPenguin321** [comment]
well this is a first

**Comment by Lumipharon** [comment]
If you had X chance to get hit from the e-sword each time you try to melee the user, that would be entertaining.

**Comment by KazeEspada** [comment]
Killed once again by being disarmed by a botanist.

**Comment by PKPenguin321** [comment]
>If you're that salty about getting disarmed, carry a taser or something next time.

¯\\\_(ツ)_/¯

**Comment by KazeEspada** [comment]
I dont see taser on my uplink. Sorry. Guess I JUST MURDERBONE WITH THE REVOLVER LIKE EVERYONE ELSE.

**Comment by Bawhoppen** [comment]
I see an ebow in my uplink

**Pull request closed without merging**